wavelink/queue.py

```diff
@@ -194,38 +202,124 @@ def _wakeup_next(self) -> None:
                 waiter.set_result(None)
                 break
 
-    def _get(self) -> Playable:
+    @staticmethod
+    def _check_compatability(item: object) -> TypeGuard[Playable]:
+        if not isinstance(item, Playable):
+            raise TypeError("This queue is restricted to Playable objects.")
+        return True
+
+    @classmethod
+    def _check_atomic(cls, item: Iterable[object]) -> TypeGuard[Iterable[Playable]]:
+        for track in item:
+            cls._check_compatability(track)
+        return True
+
+    def get(self) -> Playable:
+        """Retrieve a track from the left side of the queue. E.g. the first.
+
+        This method does not block.
+
+        .. warning::
+
+            Due to the way the queue loop works, this method will return the same track if the queue is in loop mode.
+            You can use :meth:`wavelink.Player.skip` with ``force=True`` to skip the current track.
+
+            Do **NOT** use this method to remove tracks from the queue, use either:
+
+            - ``del queue[index]``
+            - :meth:`wavelink.Queue.remove`
+            - :meth:`wavelink.Queue.delete`
+
+
+        Returns
+        -------
+        :class:`wavelink.Playable`
+            The track retrieved from the queue.
+
+        Raises
+        ------
+        QueueEmpty
+            The queue was empty when retrieving a track.
+        """
+
         if self.mode is QueueMode.loop and self._loaded:
             return self._loaded
 
         if self.mode is QueueMode.loop_all and not self:
             assert self.history is not None
 
-            self._queue.extend(self.history._queue)
+            self._items.extend(self.history._items)
             self.history.clear()
 
-        track: Playable = super()._get()
+        if not self:
```

Comment:
This could potentially extracted into a method since it's duplicated at #287 but more an observation.
e.g.
```
return self.get_loaded_track(index)
return self.get_loaded_track(0)

    def get_loaded_track(self, index: int):
        if not self:
            raise QueueEmpty("There are no items currently in this queue.")
        track: Playable = self._items.pop(index)
        self._loaded = track
        return track
```

Reply:
Not going to make this a method since it feels like if anything it would have to be private, and this logic may not be desired everywhere in the future, however I can make loaded a property.

Reply:
Made it a property with setter.